From 9a5abff3158f008ad1e7739a53dc6e680e5e7e17 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Mon, 22 Oct 2018 20:30:48 -0700
Subject: [PATCH 0482/1851] documentation: dt: Add bindings documentation for
 Streaming SCD

commit 97a075e8101a9c543239f56cbf3334bcf85fe0da from
https://github.com/Xilinx/linux-xlnx.git

This patch adds device tree binding documentation for streaming based
Scene Change Detection driver.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-scd.txt      | 50 ++++++++++++++++++-
 1 file changed, 49 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-scd.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-scd.txt
index f44a6c731e9b..a05e9712c833 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-scd.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-scd.txt
@@ -10,6 +10,13 @@ based model, which means that it will accept a dma buffer address and perform
 MEM2DEV transfer followed by statistical based image processing and give the
 data back to application if scene change detection is present or not.
 
+Another version of scene change detection IP which supports streaming model,
+which means that IP can be inserted in a capture pipeline. For example,
+"hdmirx -> streaming-scd -> fb_wr" is a typical capture pipeline where
+streaming SCD can be embedded. The IP accespts the AXI video data and perform
+histogram based statistical analysis to detect scene change. The IP supports
+single channel.
+
 Required properties:
 
 - compatible: Should be "xlnx,v-scd"
@@ -20,6 +27,10 @@ Required properties:
 
 - reset-gpios: Specifier for a GPIO that assert SCD (AP_RST_N) reset.
 
+- xlnx,memory-based: This is to differentiate between memory based and
+  streaming based IP. The value is 1 for memory based and 0 for streaming
+  based IPs.
+
 - xlnx,numstreams: Maximum active streams IP can support is 8 and this is based
   on the design.
 
@@ -31,7 +42,10 @@ Required properties:
 - port: Video port, using the DT bindings defined in ../video-interfaces.txt.
 
 Example:
-The following example shows what the device tree would look like for a memory
+
+1. Memory based device tree
+
+The following example shows how the device tree would look like for a memory
 based design where 8 streams are enabled.
 
 	scd: scenechange@a0100000 {
@@ -42,6 +56,7 @@ based design where 8 streams are enabled.
 		interrupts = <0 90 4>;
 		reset-gpios = <&gpio 94 1>;
 
+		xlnx,memory-based;
 		xlnx,numstreams = <8>;
 		xlnx,addrwidth = <0x20>;
 
@@ -114,3 +129,36 @@ based design where 8 streams are enabled.
 			};
 		};
 	};
+
+2. Streaming based device tree
+
+The following example shows how the device tree would look like for a streaming
+based design.
+
+	scd: scenechange@a0280000 {
+		compatible = "xlnx,v-scd";
+		reg = <0x0 0xa0280000 0x0 0x1fff>;
+		clocks = <&clk 72>;
+		interrupt-parent = <&gic>;
+		interrupts = <0 111 4>;
+		reset-gpios = <&gpio 100 1>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		xlnx,numstreams = <1>;
+
+		scd {
+			port@0 {
+				reg = <0x0>;
+				scd_in0: endpoint {
+					remote-endpoint = <&vpss_scaler_out>;
+				};
+			};
+
+			port@1 {
+				reg = <0x1>;
+				scd_out0: endpoint {
+					remote-endpoint = <&vcap_hdmi_in_1>;
+				};
+			};
+		};
+	};
-- 
2.31.1

